subsys/net/lib/sockets/Kconfig

```diff
@@ -96,6 +96,19 @@ config NET_SOCKETS_OFFLOAD
 	  See NET_OFFLOAD for a more deeply integrated approach which offloads
 	  from the net_context() API within the Zephyr IP stack.
 
+config NET_SOCKETS_PACKET
+	bool "Enable packet socket support"
+	default n
+	depends on NET_L2_ETHERNET
+	help
+	  This is an initial version of packet socket support (special type
+	  raw socket). Packets are passed to and from the device driver
+	  without any changes in the packet headers. It's API caller
+	  responsibility to provide all the headers (e.g L2, L3 and so on)
+	  while sending. While receiving, packets (including all the headers)
+	  will be feed to sockets as it as from the driver. This version
+	  supports only over Etherent.
```

Comment:
> This version supports only over Etherent.

This sentence also needs a grammar and typo fix up. E.g. "This version supports only Ethernet L2", or similar.

Reply:
fixing type, I don't see issue in rest.

Reply:
"supports only over" is not proper literature English. It should be "supports &lt;something> only over", or should be rephrased completely.

Reply:
Added "depends on NET_L2_ETHERNET" to config option. Hope that's fine.